src/macaron/slsa_analyzer/ci_service/github_actions/github_actions_ci.py

```diff
@@ -377,6 +382,8 @@ def workflow_run_in_date_time_range(
 
                     # Find the matching step and check its `conclusion` and `started_at` attributes.
                     html_url = None
+                    if not run_jobs["jobs"]:
```

Comment:
I believe at this point `run_jobs` is a dictionary. This if condition will only covers cases where the key `jobs` is available but its value is None or an empty dictionary. This will not cover cases where `jobs` key is not available in `run_jobs`. I wonder if this is intended ?

Reply:
I'm also not clear on the new behavior of GitHub API? Does it return an empty list now or does it completely remove the `jobs` field?

Reply:
There are multiple scenarios. GitHub removes data gradually, e.g., first the `steps` data is removed, then `jobs` data. Finally, the whole run data disappears. But the keys are not removed, instead they contain empty lists, until its parent object is removed.

Reply:
Right so we have the assumption that `jobs` and `steps` fields are always available at this point of execution.